### DISCUSSION TITLE
Site status API is always showing waiting

### DISCUSSION THREAD

**Discussion by chavda-bhavik** [plausible/analytics#4256] [Self-Hosted Support] [answered]
Hi, I've created a site on selfhosted plausible and added a script on my website. These stats is showing on my website dashboard,
![image](https://github.com/plausible/analytics/assets/50201755/19f9f729-ccde-4ae4-bee6-4e6d885e6b0c)

But api is sending a `WAITING` response, I'm trying to call this api, `GET https://plausible.app.com/api/application.com/status`

Any Idea why this error is happening? I've just switched to plausible v2 to the self-hosted community version, where I showed UI based on the website status I got from plausible.

Thank you so much for any help!

**Comment by chavda-bhavik** [comment]
Any idea regarding how to resolve this issue?

**Reply by ruslandoga** [reply]
👋 @chavda-bhavik 

You can check what SQL is executed in that request and then execute it yourself in `psql` and `clickhouse-client` to see what data is returned and how it affects status response.

**Reply by ruslandoga** [reply, marked as answer]
Here's the route: https://github.com/plausible/analytics/blob/b096057eef28f25f8223cb625dfd3303ffce521a/lib/plausible_web/router.ex#L195

Here's the controller action:
https://github.com/plausible/analytics/blob/b096057eef28f25f8223cb625dfd3303ffce521a/lib/plausible_web/controllers/api/internal_controller.ex#L8-L18

If the return value of any of these functions doesn't match what's on the left side, "WAITING" status is returned.
You can try executing these function one by one:

```console
$ cd hosting # or wherever you cloned plausible/hosting or plausible/community-edition
$ docker compose exec plausible bin/plausible remote
```
```elixir
iex> alias Plausible.{Sites, Auth, Repo}
iex> alias Plausible.Stats.Clickhouse, as: Stats
iex> alias Auth.User
# this one is different from the actual controller logic
#     %User{id: user_id} <- conn.assigns[:current_user]
# and means that the request needs to be authenticated with a cookie
# here we bypass it by loading the user directly to be able to check the other functions
iex> %User{id: user_id} = Repo.get_by(User, email: "your@email")
iex> %Site{} = site = Sites.get_by_domain("application.com")
iex> Sites.has_admin_access?(user_id, site)
true
iex> Stats.has_pageviews?(site)
true
```

**Reply by chavda-bhavik** [reply]
Thank you so much @ruslandoga for such a detailed answer.

Everything was returning true, but API was still returning `WAITING`. So I moved the application from `status` api to `/api/v1/stats/aggregate` which returns pageviews. And based on that I handled UI.

I tried Elixir the first time and it seems awesome brother. Is it possible to improve the last code snippet with the following code?
```elixir
iex> alias Plausible.{Sites, Auth}
iex> alias Plausible.Stats.Clickhouse, as: Stats
iex> alias Auth.User
# this one is different from the actual controller logic
#     %User{id: user_id} <- conn.assigns[:current_user]
# and means that the connection needs to be authenticated with a cookie
# here we bypass it by loading the user directly to be able to check the other functions
iex> %User{id: user_id} = Plausible.Repo.get_by(Plausible.Auth.User, email: "your@email")
iex> site = Sites.get_by_domain("application.com")
iex> Sites.has_admin_access?(user_id, site)
true
iex> Stats.has_pageviews?(site)
true
```
Thanks!
